### PR TITLE
[JUJU-1198] Rewrite to bash-based nw-destroy-model-lxd

### PR DESCRIPTION
**Pull request by anvial**
This patch adds destroy model test to the model suite in bash-based tests.

Original (python-based) tests live [here](https://github.com/juju/juju/blob/ac9ddcae708eca8ec2fd885a14fb27a291e424da/acceptancetests/assess_destroy_model.py).

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- ~[ ] Go unit tests, with comments saying what you're testing~
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
cd tests
./main.sh -v -p lxd model test_model_destroy
```

### CODE REVIEW

File: tests/suites/model/destroy.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	echo "Add new model 'model-new'"
+	juju add-model model-new
+
+	echo "Ensure current model is 'model-new'"
+	juju models --format json | jq -r '."current-model"' | check model-new
+
+	echo "Destroy model 'model-new'"
+	destroy_model "model-new"
+	juju switch :model-destroy
```
Comment (jack-w-shaw) [inline tests/suites/model/destroy.sh:25]: We technically don't need the colon here

And also, this is incorrect I believe. The original test asserts that if the active model is destroyed, you are automatically switched to another

File: tests/suites/model/destroy.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	echo "Ensure current model is 'model-destroy'"
+	juju models --format json | jq -r '."current-model"' | check model-destroy
+
+	destroy_model "model-destroy"
+}
```
Comment (jack-w-shaw) [inline tests/suites/model/destroy.sh:31]: One small bit of coverage missing
https://github.com/juju/juju/blob/ac9ddcae708eca8ec2fd885a14fb27a291e424da/acceptancetests/assess_destroy_model.py#L68-L70

The python based test ensures that the created model can't be found

**Review by jack-w-shaw — CHANGES_REQUESTED** [review]

File: tests/suites/model/destroy.sh (hunk 1 above)
Comment (anvial) [reply tests/suites/model/destroy.sh:25]: I agreed about the colon. But I've concerns about automatic switching: 
https://pastebin.canonical.com/p/99Y9R7jQ7b/

Here is the origin (python-based) test: https://github.com/juju/juju/blob/137a772ed339b73b856e9adc0a5624976c2890b2/acceptancetests/assess_destroy_model.py#L47-L53

File: tests/suites/model/destroy.sh (hunk 2 above)
Comment (anvial) [reply tests/suites/model/destroy.sh:34]:
```sh
destroy_model "model-new"
is_destroyed=$(juju models --format json | jq -r '.models[] | select(."short-name" == "model-new")')
if [[ -z ${is_destroyed} ]]; then is_destroyed=true; fi
check_contains "${is_destroyed}" true
```
Added explicit model deletion check.

**Review by jack-w-shaw — APPROVED** [review]
LGTM

**Pull request merged**